common/include/pcl/common/impl/file_io.hpp

```diff
@@ -42,29 +42,21 @@
 namespace pcl
 {
 
-#ifndef _WIN32
   void getAllPcdFilesInDirectory(const std::string& directory, std::vector<std::string>& file_names)
   {
-    DIR *dp;
-    struct dirent *dirp;
-    if((dp  = opendir(directory.c_str())) == nullptr) {
-      std::cerr << "Could not open directory.\n";
-      return;
+    boost::filesystem::path p(directory);
```

Comment:
We should clear the file_names at the beginning, right?

Reply:
There could be cases where the user reuses the same vector to get a list of the PCD files from multiple directories. So not sure if clearing is a good idea.

Reply:
In that case, sorting is a needless expense for empty directory but non-empty vector.

I did a grep and there's no usage of this function in PCL code, and no test; so I can't guess how this is supposed to be used. The docstring makes no promises regarding use across multi dir (but it makes sense).

Reply:
Yeah true, we can make sorting to be done only if non-empty directory. But do we really need sorting at all? 
For most cases, I don't think it is needed and when needed the user can do it on his side easily.

Reply:
We don't say sorted vector, so I think we can just return it without sorting. @taketwo Is this ok? This doesn't technically require us to add the "changes behavior" label

Reply:
In the docstring, it is mentioned that it sorts the vector. So I guess it is changing behavior?

Reply:
To be honest, dunno what's better, to follow actual behavior or claimed behavior.

Reply:
We could add another argument sorted with the default value set to true. This will allow existing behavior to not break.